app/src/ereferrals/java/org/eyeseetea/malariacare/data/sync/importer/strategies/PullControllerStrategy.java

```diff
@@ -73,12 +76,20 @@ public void onPullDataComplete(final IPullController.Callback callback, boolean
                         orgUnitDataSource.getUserOrgUnit(
                                 credentialsLocalDataSource.getOrganisationCredentials());
                 org.eyeseetea.malariacare.domain.entity.Program program = orgUnit.getProgram();
-                programLocalDataSource.saveUserProgramId(program);
+                saveProgram(programLocalDataSource, program);
             } catch (Exception e) {
                 e.printStackTrace();
                 callback.onError(e);
             }
             mPullController.convertData(callback);
         }
     }
+
+    private void saveProgram(IProgramRepository programLocalDataSource, Program program) {
+        if(!programLocalDataSource.checkLastDownloadedProgramMedia(program.getCode())){
```

Comment:
I think that program associate to media should be program associated to the user then saves this info in preferences is not necessary and remove media files is not a ProgramDataSource responsibility, from here should remove media local folder directly using SDK FileUtils because this class is not a domain class